test/nsqdconnection_test.js

```diff
@@ -229,7 +229,7 @@ describe('WriterConnectionState', () => {
     const { statemachine, connection } = state;
 
     connection.on(WriterNSQDConnection.READY, () => {
-      connection.produceMessages('test', ['one'], () => done());
+      connection.produceMessages('test', ['one'], undefined, () => done());
```

Comment:
Do you really need to provide the undefined here if there's logic to deal wit the fact that `timeMs` is optional?